src/providers/google-reverse-geocoder.php

```diff
@@ -2,11 +2,58 @@
 
 namespace SearchApi\Providers;
 
-use SearchApi\Services\ReverseGeocoder;
+use SearchApi\Models as Models;
+use SearchApi\Support as Support;
+use SearchApi\Test\Support\Mocks as Mocks;
+use SearchApi\Services\ReverseGeocoder as ReverseGeocoder;
 
 /**
- * Class GoogleReverseGeocoder - Given a location name find a point or polygon that can represent that
+ * Class GoogleReverseGeocoder - Given a latlin or place id to find a point or polygon that can represent that
  * location via GoogleMaps.
  */
 class GoogleReverseGeocoder implements ReverseGeocoder {
+  public $api_key; // service key
+  private $geo_parser; // injected parser
+
+  public function __construct( $key = null, $parser = null ) {
+  	$this->api_key = $key;
+
+  	if ( $parser === null ) {
```

Comment:
Ill check if there a way to tell the ide i am using to use space when I hit enter... I think that is the issue happening with these tabs.